documentation/plan_functions.md

```diff
@@ -0,0 +1,1646 @@
+# Bolt functions and types
```

Comment:
I personally like keeping types and functions separate, following the Puppet documentation and avoiding a super long multi-purpose doc.

Reply:
Yeah, I prefer the option of a separate doc for types